README.md

```diff
@@ -37,6 +37,148 @@ This is an implementation of [SD-JWT (I-D version 05)](https://www.ietf.org/arch
   - [ ] Documentation
   - [ ] Publish on npm
 
+## Disclosure Frame
+To issue or pack claims into a valid SD-JWT we use Disclosure Frame to define which properties/values should be selectively diclosable. \
+It follows the following format:
+```typescript
+type ArrayIndex = number;
+type DisclosureFrame = {
+  [key: string | ArrayIndex]: DisclosureFrame | unknown;
+  _sd?: Array<string | ArrayIndex>;
+};
+```
+### Examples:
+
+#### set property as selectively disclosable
+```js
+const claims = {
+  firstname: 'John',
+  lastname: 'Doe'
+}
+const diclosureFrame = {
+  _sd: ['firstname'] // set firstname as selectively discloseable
+}
+
+// result
+const sdjwt = {
+  _sd: ['LjgwZy8TNXmmPO9mNqVDtq3jiX5r3YS-P-qw2hBNYyU']
+  lastname: 'Doe',
+}
+```
+
+#### nested property
+```js
+const claims = {
+  address: {
+    street: '123 Main St',
+    suburb: 'Anytown',
+    postcode: '1234'
+  }
+}
+
+const disclosureFrame = {
+  address: {
+    // set address.street and address.suburb as selectively discloseable
+    _sd: ['street', 'suburb'];
+  }
+}
+
+// result
+const sdjwt = {
+  address: {
+    _sd: [
+      '02d7bUYevjfAzJ0Gr42ymHy66ezQVL7huNGBO68xSfs',
+      'ai7P4vgPZ-Jk1QwL55BLQqtN2gwWy31-pi2VGWiIggs',
+    ],
+    postcode: '1234'
+  }
+}
+```
+
+#### Array item
+```js
+const claims = {
+  nicknames: ['Johnny', 'JD']
+}
+
+const disclosureFrame = {
+  nicknames: {
+    _sd: [0, 1] // index of items in 'nicknames' Array
+  }
+}
+
+// result
+const sdjwt = {
+  nicknames: [
+    { '...': 'yfhdm_aKTMgm666j79GoXr2mer2dBW0cFfap8iXnAzY' },
+    { '...': 'EU0ORASnAlqNtRwttBXsGTISxQ6myFPMBHPE0Ds8aSE' }
+  ]
+}
+```
+
+#### Object in Arrays
+```js
+const claims = {
+  items: [
+    {
+      type: 'shirt',
+      size: 'M'
+    },
+    'Towel',
+    'Water Bottle'
+  ]
+}
+
+const disclosureFrame = {
+  items: {
```

Comment:
Hm... Isn't it a little strange that in one case marking array items as disclosable you use `[]` notation and here it is `{}`?
Let's not change anything yet, but might be worth creating a ticket for future.